// @googleforcreators/element-library/dist/media/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("styled-components"),t=require("../shared/shared.js");require("../shared/layerText.js"),require("@googleforcreators/patterns"),require("@googleforcreators/transform");var o=require("./constants.js");function r(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}const s=r(e).default.div.withConfig({displayName:"media__CropBox",componentId:"sc-ttu8xg-0"})(["width:100%;height:100%;position:relative;overflow:hidden;"," &::after{content:'';display:block;position:absolute;left:0;top:0;width:100%;height:100%;pointer-events:none;}"],t.elementWithBorderRadius);exports.MEDIA_DEFAULT_ATTRIBUTES=o.MEDIA_DEFAULT_ATTRIBUTES,exports.MEDIA_MASK_OPACITY=o.MEDIA_MASK_OPACITY,exports.MEDIA_PANELS=o.MEDIA_PANELS,exports.canFlip=o.canFlip,exports.editModeGrayout=o.editModeGrayout,exports.hasEditMode=o.hasEditMode,exports.isMaskable=o.isMaskable,exports.isMedia=o.isMedia,exports.resizeRules=o.resizeRules,exports.CropBox=s;
