// @googleforcreators/design-system/dist/components/snackbar/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";function t(t){return t&&"object"==typeof t&&"default"in t?t:{default:t}}Object.defineProperty(exports,"__esModule",{value:!0});var e=t(require("prop-types"));e.default.oneOf(["top-left","bottom-left","top-right","bottom-right","top","bottom"]);const o={SUCCESS:"success",ERROR:"error"},s=e.default.shape({src:e.default.string.isRequired,alt:e.default.string,status:e.default.oneOf(Object.values(o))});e.default.shape({message:e.default.oneOfType([e.default.node,e.default.string]),actionLabel:e.default.oneOfType([e.default.node,e.default.string]),actionHelpText:e.default.string,dismissible:e.default.bool,onAction:e.default.func,onDismiss:e.default.func,preventActionDismiss:e.default.bool,preventAutoDismiss:e.default.bool,timeout:e.default.number,thumbnail:s,id:e.default.string}),exports.AUTO_REMOVE_MESSAGE_TIME_INTERVAL_MAX=1e4,exports.AUTO_REMOVE_MESSAGE_TIME_INTERVAL_MIN=1200,exports.DEFAULT_MESSAGE_Z_INDEX=15,exports.SnackbarNotificationThumbnail=s,exports.THUMBNAIL_STATUS=o;
