// @googleforcreators/design-system/dist/components/contextMenu/components/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./button.js"),r=require("./group.js"),u=require("./icon.js"),s=require("./item.js"),t=require("./suffix.js"),o=require("./label.js"),i=require("./link.js"),n=require("./separator.js"),p=require("./shortcut.js"),x=require("./subMenuTrigger.js");exports.MenuButton=e,exports.MenuGroup=r,exports.MenuIcon=u,exports.MenuItem=s,exports.MenuItemSuffix=t,exports.MenuLabel=o,exports.MenuLink=i,exports.MenuSeparator=n,exports.MenuShortcut=p,exports.SubMenuTrigger=x;
