// @googleforcreators/patterns/dist/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./types.js"),r=require("./convertToCSS.js"),t=require("./createSolid.js"),o=require("./createSolidFromString.js"),s=require("./generatePatternStyles.js"),i=require("./getHexFromSolid.js"),p=require("./getHexFromSolidArray.js"),a=require("./getHexFromValue.js"),x=require("./getOpaquePattern.js"),u=require("./getPreviewText.js"),l=require("./getSolidFromHex.js"),S=require("./hasGradient.js"),n=require("./hasOpacity.js"),q=require("./isPatternEqual.js"),g=require("./isHexColorString.js");exports.ColorStopPropType=e.ColorStopPropType,exports.HexPropType=e.HexPropType,exports.PatternPropType=e.PatternPropType,exports.convertToCSS=r,exports.createSolid=t,exports.createSolidFromString=o,exports.generatePatternStyles=s,exports.getHexFromSolid=i,exports.getHexFromSolidArray=p,exports.getHexFromValue=a,exports.getOpaquePattern=x,exports.getPreviewText=u,exports.getSolidFromHex=l,exports.hasGradient=S,exports.hasOpacity=n,exports.isPatternEqual=q,exports.isHexColorString=g;
