// @googleforcreators/element-library/dist/text/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/patterns"),r=require("@googleforcreators/design-system"),o=require("@googleforcreators/elements"),s=require("../shared/constants.js");require("../shared/shared.js"),require("../shared/layerText.js"),require("@googleforcreators/transform");const t={...s.SHARED_DEFAULT_ATTRIBUTES,backgroundTextMode:o.BACKGROUND_TEXT_MODE.NONE,font:o.TEXT_ELEMENT_DEFAULT_FONT,fontSize:36,backgroundColor:e.createSolid(196,196,196),lineHeight:1.3,textAlign:"initial",padding:{vertical:0,horizontal:0,locked:!0}},a=[r.PanelTypes.ELEMENT_ALIGNMENT,r.PanelTypes.SIZE_POSITION,r.PanelTypes.TEXT_STYLE,r.PanelTypes.BORDER,r.PanelTypes.LINK,r.PanelTypes.ANIMATION];exports.canFlip=!1,exports.defaultAttributes=t,exports.hasEditMode=!0,exports.hasEditModeMoveable=!0,exports.isMaskable=!1,exports.isMedia=!1,exports.panels=a,exports.resizeRules={vertical:!1,horizontal:!0,diagonal:!0,minWidth:20,minHeight:0};
