// @googleforcreators/element-library/dist/text/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./display.js"),t=require("./edit.js"),s=require("./frame.js"),r=require("./output.js"),i=require("./textContent.js"),o=require("./layer.js"),a=require("./icon.js"),p=require("./updateForResizeEvent.js"),u=require("./constants.js");exports.Display=e,exports.Edit=t,exports.Frame=s,exports.Output=r.default,exports.TextContent=i,exports.LayerContent=o,exports.LayerIcon=a,exports.updateForResizeEvent=p,exports.canFlip=u.canFlip,exports.defaultAttributes=u.defaultAttributes,exports.hasEditMode=u.hasEditMode,exports.hasEditModeMoveable=u.hasEditModeMoveable,exports.isMaskable=u.isMaskable,exports.isMedia=u.isMedia,exports.panels=u.panels,exports.resizeRules=u.resizeRules;
