// @googleforcreators/stickers/dist/pizzas-in-nyc/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./tomatoes.js"),r=require("./chili.js"),s=require("./onion.js"),i=require("./veggies.js"),t=require("./arrowRight.js"),o=require("./cheese.js"),u=require("./thickBrushStroke.js"),c=require("./mediumBrushStroke.js"),h=require("./tiltedBrushStroke.js");exports.nycTomatoes=e,exports.nycChili=r,exports.nycOnion=s,exports.nycVeggies=i,exports.nycArrowRight=t,exports.nycCheese=o,exports.nycThickBrush=u,exports.nycMediumBrush=c,exports.nycTiltedBrush=h;
