// @googleforcreators/story-editor/dist/app/media/media3p/actions.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./categories/actions.js"),t=require("../pagination/actions.js"),r=require("./types.js");exports.deselectCategory=e.deselectCategory,exports.fetchCategoriesError=e.fetchCategoriesError,exports.fetchCategoriesStart=e.fetchCategoriesStart,exports.fetchCategoriesSuccess=e.fetchCategoriesSuccess,exports.selectCategory=e.selectCategory,exports.deleteMediaElement=t.deleteMediaElement,exports.fetchMediaError=t.fetchMediaError,exports.fetchMediaStart=t.fetchMediaStart,exports.fetchMediaSuccess=t.fetchMediaSuccess,exports.setNextPage=t.setNextPage,exports.updateMediaElement=t.updateMediaElement,exports.setSearchTerm=e=>t=>{let{searchTerm:s}=t;e({type:r.MEDIA3P_SET_SEARCH_TERM,payload:{searchTerm:s}})},exports.setSelectedProvider=e=>t=>{let{provider:s}=t;e({type:r.MEDIA3P_SET_SELECTED_PROVIDER,payload:{provider:s}})};
