// @googleforcreators/rich-text/dist/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./provider.js"),t=require("./editor.js"),r=require("./context.js"),s=require("./useRichText.js"),o=require("./usePasteTextContent.js"),i=require("./getFontVariants.js"),a=require("./utils/getCaretCharacterOffsetWithin.js"),x=require("./htmlManipulation.js");exports.RichTextProvider=e,exports.RichTextEditor=t,exports.RichTextContext=r,exports.useRichText=s,exports.usePasteTextContent=o,exports.getFontVariants=i,exports.getCaretCharacterOffsetWithin=a,exports.getHTMLFormatters=x.getHTMLFormatters,exports.getHTMLInfo=x.getHTMLInfo,exports.getSelectAllStateFromHTML=x.getSelectAllStateFromHTML;
