// @googleforcreators/stickers/dist/skin-care-at-home/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./violetLotus.js"),s=require("./sleepMask.js"),r=require("./toiletries.js"),o=require("./combAndScissors.js"),i=require("./towelTurban.js"),t=require("./palmIcon.js");exports.skinCareFlower=e,exports.skinCareSleepMask=s,exports.skinCareFaceWashBottles=r,exports.skinCareCombAndScissors=o,exports.skinCareFaceIcon=i,exports.skinCarePalmIcon=t;
