// @googleforcreators/story-editor/dist/app/media/pagination/actions.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./types.js");exports.deleteMediaElement=t=>r=>{let{provider:a,id:p}=r;t({type:e.DELETE_MEDIA_ELEMENT,payload:{provider:a,id:p}})},exports.fetchMediaError=t=>r=>{let{provider:a}=r;t({type:e.FETCH_MEDIA_ERROR,payload:{provider:a}})},exports.fetchMediaStart=t=>r=>{let{provider:a,pageToken:p}=r;t({type:e.FETCH_MEDIA_START,payload:{provider:a,pageToken:p}})},exports.fetchMediaSuccess=t=>r=>{let{provider:a,media:p,nextPageToken:d,totalPages:o,...i}=r;t({type:e.FETCH_MEDIA_SUCCESS,payload:{provider:a,media:p,nextPageToken:d,totalPages:o,...i}})},exports.setNextPage=t=>r=>{let{provider:a}=r;t({type:e.SET_NEXT_PAGE,payload:{provider:a}})},exports.updateMediaElement=t=>r=>{let{provider:a,id:p,data:d}=r;t({type:e.UPDATE_MEDIA_ELEMENT,payload:{provider:a,id:p,data:d}})};
