// @googleforcreators/masks/dist/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./masks.js"),r=require("./constants.js"),s=require("./utils/elementBorder.js"),t=require("./display.js"),o=require("./frame.js"),p=require("./output.js");exports.canMaskHaveBorder=e.canMaskHaveBorder,exports.canSupportMultiBorder=e.canSupportMultiBorder,exports.getBorderedMaskProperties=e.getBorderedMaskProperties,exports.getElementMask=e.getElementMask,exports.getMaskByType=e.getMaskByType,exports.DEFAULT_MASK=r.DEFAULT_MASK,exports.MASKS=r.MASKS,exports.MaskTypes=r.MaskTypes,exports.getBorderColor=s.getBorderColor,exports.getBorderPositionCSS=s.getBorderPositionCSS,exports.getBorderRadius=s.getBorderRadius,exports.getBorderStyle=s.getBorderStyle,exports.getResponsiveBorder=s.getResponsiveBorder,exports.shouldDisplayBorder=s.shouldDisplayBorder,exports.DisplayWithMask=t,exports.FrameWithMask=o,exports.OutputWithMask=p;
