// @googleforcreators/design-system/dist/theme/helpers/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./outline.js"),r=require("./scrollbar.js"),t=require("./expandPresetStyles.js"),s=require("./fullSize.js"),l=require("./centerContent.js"),n=require("./visuallyHidden.js"),o=require("./transparentBg.js");exports.focusCSS=e.focusCSS,exports.focusableOutlineCSS=e.focusableOutlineCSS,exports.SCROLLBAR_WIDTH=r.SCROLLBAR_WIDTH,exports.scrollbarCSS=r.scrollbarCSS,exports.expandPresetStyles=t.expandPresetStyles,exports.expandTextPreset=t.expandTextPreset,exports.fullSizeAbsolute=s.fullSizeAbsolute,exports.fullSizeRelative=s.fullSizeRelative,exports.centerContent=l.centerContent,exports.visuallyHidden=n.visuallyHidden,exports.transparentBg=o.transparentBg;
