// @googleforcreators/stickers/dist/art-books-gift-guide/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./books.js"),r=require("./photoFrame.js"),a=require("./openHands.js"),o=require("./handHeldSign.js"),s=require("./manReading.js"),t=require("./womanReading.js"),n=require("./womanReading02.js");exports.artBooks=e,exports.artPhotoFrame=r,exports.artHands=a,exports.artHandHeldSign=o,exports.artManReading=s,exports.artWomanReading=t,exports.artWomanReading02=n;
