// @googleforcreators/element-library/dist/gif/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/design-system"),r=require("../shared/constants.js");require("../shared/shared.js"),require("../shared/layerText.js"),require("@googleforcreators/patterns"),require("@googleforcreators/transform"),require("../media/index.js");var s=require("../media/constants.js");const t={...r.SHARED_DEFAULT_ATTRIBUTES,...s.MEDIA_DEFAULT_ATTRIBUTES},a=[e.PanelTypes.ELEMENT_ALIGNMENT,...s.MEDIA_PANELS,e.PanelTypes.LINK,e.PanelTypes.IMAGE_ACCESSIBILITY];exports.canFlip=s.canFlip,exports.editModeGrayout=s.editModeGrayout,exports.hasEditMode=s.hasEditMode,exports.isMaskable=s.isMaskable,exports.isMedia=s.isMedia,exports.resizeRules=s.resizeRules,exports.defaultAttributes=t,exports.panels=a;
