// @googleforcreators/element-library/dist/video/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/design-system"),s=require("../media/constants.js"),r=require("../shared/constants.js");require("../shared/shared.js"),require("../shared/layerText.js"),require("@googleforcreators/patterns"),require("@googleforcreators/transform");const t={...s.resizeRules},o={...r.SHARED_DEFAULT_ATTRIBUTES,...s.MEDIA_DEFAULT_ATTRIBUTES,controls:!1,loop:!1,autoPlay:!0,tracks:[],resource:{posterId:null,poster:null,id:0,alt:""}},a=[e.PanelTypes.ELEMENT_ALIGNMENT,...s.MEDIA_PANELS,e.PanelTypes.VIDEO_OPTIONS,e.PanelTypes.VIDEO_ACCESSIBILITY,e.PanelTypes.CAPTIONS];exports.canFlip=s.canFlip,exports.editModeGrayout=s.editModeGrayout,exports.hasEditMode=s.hasEditMode,exports.isMaskable=s.isMaskable,exports.isMedia=s.isMedia,exports.defaultAttributes=o,exports.panels=a,exports.resizeRules=t;
