// @googleforcreators/stickers/dist/a-day-in-the-life/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./brownRice.js"),r=require("./fish.js"),s=require("./yellowLeaves.js"),i=require("./jumpRope.js"),o=require("./dumbbells.js"),a=require("./whiteBag.js"),u=require("./alarmClock.js");exports.dayInTheLifeBrownRice=e,exports.dayInTheLifeFish=r,exports.dayInTheLifeLeaves=s,exports.dayInTheLifeJumpRope=i,exports.dayInTheLifeDumbells=o,exports.dayInTheLifeBag=a,exports.dayInTheLifeAlarmClock=u;
