// @googleforcreators/story-editor/dist/app/media/local/types.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});exports.LOCAL_MEDIA_ADD_AUDIO_PROCESSING="LOCAL_MEDIA_ADD_AUDIO_PROCESSING",exports.LOCAL_MEDIA_ADD_BASE_COLOR_PROCESSING="LOCAL_MEDIA_ADD_BASE_COLOR_PROCESSING",exports.LOCAL_MEDIA_ADD_BLURHASH_PROCESSING="LOCAL_MEDIA_ADD_BLURHASH_PROCESSING",exports.LOCAL_MEDIA_ADD_POSTER_PROCESSING="LOCAL_MEDIA_ADD_POSTER_PROCESSING",exports.LOCAL_MEDIA_PREPEND_MEDIA="LOCAL_MEDIA_PREPEND_MEDIA",exports.LOCAL_MEDIA_REMOVE_AUDIO_PROCESSING="LOCAL_MEDIA_REMOVE_AUDIO_PROCESSING",exports.LOCAL_MEDIA_REMOVE_BASE_COLOR_PROCESSING="LOCAL_MEDIA_REMOVE_BASE_COLOR_PROCESSING",exports.LOCAL_MEDIA_REMOVE_BLURHASH_PROCESSING="LOCAL_MEDIA_REMOVE_BLURHASH_PROCESSING",exports.LOCAL_MEDIA_REMOVE_POSTER_PROCESSING="LOCAL_MEDIA_REMOVE_POSTER_PROCESSING",exports.LOCAL_MEDIA_RESET_FILTERS="LOCAL_MEDIA_RESET_FILTERS",exports.LOCAL_MEDIA_SET_MEDIA="LOCAL_MEDIA_SET_MEDIA",exports.LOCAL_MEDIA_SET_MEDIA_TYPE="LOCAL_MEDIA_SET_MEDIA_TYPE",exports.LOCAL_MEDIA_SET_SEARCH_TERM="LOCAL_MEDIA_SET_SEARCH_TERM",exports.LOCAL_MEDIA_TYPE_ALL="LOCAL_MEDIA_TYPE_ALL";
