// @googleforcreators/element-library/dist/product/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/patterns"),r=require("@googleforcreators/design-system"),s=require("../shared/constants.js");const o=e.createSolidFromString("#c4c4c4"),t={...s.SHARED_DEFAULT_ATTRIBUTES,backgroundColor:o,lockDimensions:!0,supportsRotation:!1},i=[r.PanelTypes.PRODUCT];exports.canFlip=!0,exports.defaultAttributes=t,exports.hasEditMode=!1,exports.isMaskable=!1,exports.isMedia=!1,exports.panels=i,exports.resizeRules={vertical:!0,horizontal:!0,diagonal:!0,minWidth:20,minHeight:20};
