// @googleforcreators/units/dist/dimensions.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var t=require("./constants.js");function e(t){return Number(t.toFixed(0))}function o(t){return Number(t.toFixed(5))}function r(e,r){return o(e*r/t.PAGE_WIDTH)}function n(e,r){return o(e*r/t.PAGE_HEIGHT)}exports.dataFontEm=function(e){return Number((e*t.DEFAULT_EM).toFixed(1))},exports.dataPixels=e,exports.dataToEditorX=r,exports.dataToEditorY=n,exports.dataToFontSizeY=function(t,e){return(n(t,e)/10).toFixed(6)},exports.editorPixels=o,exports.editorToDataX=function(o,r){let n=!(arguments.length>2&&void 0!==arguments[2])||arguments[2];const i=o*t.PAGE_WIDTH/r;return!1===n?i:e(i)},exports.editorToDataY=function(o,r){let n=!(arguments.length>2&&void 0!==arguments[2])||arguments[2];const i=o*t.PAGE_HEIGHT/r;return!1===n?i:e(i)},exports.getBox=function(e,o,i){let{x:u,y:d,width:s,height:a,rotationAngle:x,isBackground:E}=e;return{x:r(E?0:u,o),y:n(E?-t.DANGER_ZONE_HEIGHT:d,i),width:r(E?t.PAGE_WIDTH:s,o),height:n(E?t.FULLBLEED_HEIGHT:a,i),rotationAngle:E?0:x}};
