// @googleforcreators/design-system/dist/components/keyboard/gridview/utils.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0}),exports.focusOnPage=function(t){const n=t?.querySelectorAll(["button","a"])?.[0];n&&n.focus()},exports.getArrowDir=function(t,n,e,o){const r=o?-1:1;return t===n?r:t===e?-1*r:0},exports.getColumn=function(t,n){return t%n+1},exports.getGridColumnAndRowCount=function(t,n){let e,o=0;for(const n of t.children){const{x:t}=n.getBoundingClientRect();if(null!=e&&t<e)break;e=t,o++}return{rows:Math.ceil(n/o),columns:o}},exports.getIndex=function(t){let{row:n,column:e,numRows:o,numColumns:r,numItems:u}=t;if(n>o||n<=0||e>r||e<=0)return-1;const c=r*(n-1)+(e-1);return c>=u?u-1:c},exports.getRow=function(t,n){return Math.ceil((t+1)/n)};
