// @googleforcreators/stickers/dist/doers-get-more-done/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var r=require("./facebookIcon.js"),e=require("./instagramIcon.js"),o=require("./orangeCross.js"),s=require("./orangeDot.js"),t=require("./plus.js"),i=require("./rightArrow.js"),u=require("./twitterIcon.js"),n=require("./youTubeIcon.js");exports.diyFacebookIcon=r,exports.diyInstagramIcon=e,exports.diyOrangeCross=o,exports.diyOrangeDot=s,exports.diyPlus=t,exports.diyRightArrow=i,exports.diyTwitterIcon=u,exports.diyYouTubeIcon=n;
