// @googleforcreators/story-editor/dist/components/footer/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/design-system");const _=e.THEME_CONSTANTS.ICON_SIZE+6+2,E=2*(e.THEME_CONSTANTS.LARGE_BUTTON_SIZE+2)+_+16+16;exports.FOOTER_BOTTOM_MARGIN=16,exports.FOOTER_MARGIN=16,exports.FOOTER_MENU_GAP=8,exports.FOOTER_TOP_MARGIN=16,exports.KEYBOARD_SHORTCUTS_PADDING=3,exports.MAX_MENU_WIDTH=E,exports.VERY_WIDE_MARGIN=8,exports.VERY_WIDE_WORKSPACE_LIMIT=1e3;
