// @googleforcreators/tracking/dist/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var r=require("./initializeTracking.js"),e=require("./enableTracking.js"),i=require("./disableTracking.js"),t=require("./isTrackingEnabled.js"),a=require("./trackClick.js"),s=require("./trackError.js"),c=require("./trackEvent.js"),k=require("./trackTiming.js"),n=require("./trackScreenView.js"),o=require("./getTimeTracker.js");exports.initializeTracking=r,exports.enableTracking=e,exports.disableTracking=i,exports.isTrackingEnabled=t,exports.trackClick=a,exports.trackError=s,exports.trackEvent=c,exports.trackTiming=k,exports.trackScreenView=n,exports.getTimeTracker=o;
