// @googleforcreators/story-editor/dist/components/colorPicker/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var F=require("@googleforcreators/patterns");const E=E=>F.createSolidFromString(`#${E}`),A=[E("00000000"),E("000000"),E("454545"),E("B6B6B6"),E("EEEEEE"),E("FFFFFF"),E("003BAD"),E("0044C9"),E("0354F2"),E("1263FF"),E("347EFF"),E("6D9FFF"),E("4B1283"),E("5A189A"),E("7B2CBF"),E("9D4EDD"),E("C77DFF"),E("E0AAFF"),E("E03568"),E("FF477E"),E("FF5C8A"),E("FF7096"),E("FF85A1"),E("FF99AC"),E("590D22"),E("800F2F"),E("A4133C"),E("C9184A"),E("FF4D6D"),E("FF758F"),E("DBCA09"),E("FFD900"),E("FFE747"),E("FFEE70"),E("FFF599"),E("FFFFB7"),E("E06E04"),E("EA7D00"),E("FF9500"),E("FFA811"),E("FFBC35"),E("FFCC4A"),E("607047"),E("79895D"),E("839568"),E("A5B98B"),E("C1D3AA"),E("D4E4C0"),E("603809"),E("603808"),E("6F4518"),E("8B5E34"),E("A47148"),E("BC8A5F")];exports.BASIC_COLORS=A,exports.CONFIRMATION_DIALOG_STORAGE_KEY="DELETE_COLOR_PRESET_DIALOG_DISMISSED",exports.GRADIENT_STOP_SIZE=14,exports.LINE_LENGTH=112,exports.LINE_WIDTH=12,exports.TYPE_LINEAR="linear",exports.TYPE_RADIAL="radial",exports.TYPE_SOLID="solid";
