// @googleforcreators/element-library/dist/sticker/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/design-system"),s=require("../shared/constants.js");require("../shared/shared.js"),require("../shared/layerText.js"),require("@googleforcreators/patterns"),require("@googleforcreators/transform");const r=s.SHARED_DEFAULT_ATTRIBUTES,t=[e.PanelTypes.ELEMENT_ALIGNMENT,e.PanelTypes.SIZE_POSITION,e.PanelTypes.ANIMATION,e.PanelTypes.LINK];exports.canFlip=!0,exports.defaultAttributes=r,exports.hasEditMode=!1,exports.isAspectAlwaysLocked=!0,exports.isMaskable=!1,exports.isMedia=!1,exports.panels=t,exports.resizeRules={diagonal:!0,minWidth:20,minHeight:20};
