// @googleforcreators/dashboard/dist/constants/pageStructure.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/units");const t={TABLET:"tablet",DESKTOP:"desktop",THUMBNAIL:"thumbnail"},r={[t.TABLET]:200,[t.DESKTOP]:232,[t.THUMBNAIL]:33};Object.defineProperty(exports,"PAGE_HEIGHT",{enumerable:!0,get:function(){return e.PAGE_HEIGHT}}),Object.defineProperty(exports,"PAGE_RATIO",{enumerable:!0,get:function(){return e.PAGE_RATIO}}),Object.defineProperty(exports,"PAGE_WIDTH",{enumerable:!0,get:function(){return e.PAGE_WIDTH}}),exports.DASHBOARD_LEFT_NAV_WIDTH=288,exports.GRID_SPACING={COLUMN_GAP:24,ROW_GAP:32},exports.MIN_DASHBOARD_WIDTH=1098,exports.PAGE_WRAPPER={GUTTER:52},exports.STORY_PREVIEW_WIDTH=r,exports.VIEWPORT_BREAKPOINT=t,exports.WPBODY_ID="wpbody";
