// @googleforcreators/story-editor/dist/components/footer/carousel/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});const T=require("@googleforcreators/design-system").THEME_CONSTANTS.ICON_SIZE,_=T+8;exports.DRAWER_BUTTON_GAP_COLLAPSED=2,exports.DRAWER_BUTTON_GAP_EXPANDED=8,exports.DRAWER_BUTTON_HEIGHT=32,exports.NARROW_THUMBNAIL_HEIGHT=54,exports.NARROW_THUMBNAIL_WIDTH=36,exports.NAVIGATION_BUTTON_GAP=8,exports.NAVIGATION_BUTTON_GUTTER=_,exports.NAVIGATION_BUTTON_WIDTH=T,exports.THUMBNAIL_LINE_WIDTH=4,exports.THUMBNAIL_MARGIN=16,exports.WIDE_CAROUSEL_LIMIT=400,exports.WIDE_THUMBNAIL_HEIGHT=60,exports.WIDE_THUMBNAIL_WIDTH=40;
