// @googleforcreators/element-library/dist/shape/constants.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("@googleforcreators/patterns"),r=require("@googleforcreators/design-system"),s=require("../shared/constants.js");require("../shared/shared.js"),require("../shared/layerText.js"),require("@googleforcreators/transform");const o=e.createSolidFromString("#c4c4c4"),t={...s.SHARED_DEFAULT_ATTRIBUTES,backgroundColor:o},a=[r.PanelTypes.COLOR_PRESETS,r.PanelTypes.ELEMENT_ALIGNMENT,r.PanelTypes.SIZE_POSITION,r.PanelTypes.BORDER,r.PanelTypes.LINK,r.PanelTypes.SHAPE_STYLE,r.PanelTypes.ANIMATION];exports.canFlip=!0,exports.defaultAttributes=t,exports.hasEditMode=!1,exports.isMaskable=!0,exports.isMedia=!1,exports.panels=a,exports.resizeRules={vertical:!0,horizontal:!0,diagonal:!0,minWidth:20,minHeight:20};
