// @googleforcreators/element-library/dist/shape/index.js
/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./display.js"),s=require("./output.js"),t=require("./layer.js"),r=require("./icon.js"),i=require("./constants.js");exports.Display=e,exports.Output=s,exports.LayerContent=t,exports.LayerIcon=r,exports.canFlip=i.canFlip,exports.defaultAttributes=i.defaultAttributes,exports.hasEditMode=i.hasEditMode,exports.isMaskable=i.isMaskable,exports.isMedia=i.isMedia,exports.panels=i.panels,exports.resizeRules=i.resizeRules;
